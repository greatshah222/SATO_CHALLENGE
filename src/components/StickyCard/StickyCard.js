import PropTypes from 'prop-types';
import React from 'react';
import Sticky from 'react-stickynode';
import { withStyles } from '@material-ui/core';
import classNames from 'classnames';
import CallToActionArea from '../CallToActionArea';
import * as designTokens from '../../.generated/styles';

const styles = theme => ({
  animContainer: {
    transition: 'padding .5s, margin .5s, border-radius .5s, height 5s',
  },
  animContainerNormal: {
    [theme.breakpoints.down('xs')]: {
      marginLeft: designTokens.spacingMedium,
      marginRight: designTokens.spacingMedium,
    },
  },
  animContainerSticky: {
    padding: `${designTokens.spacingMedium} calc(10vw + ${designTokens.spacingXl})`,
    borderRadius: '0',
    boxShadow: designTokens.shadowHigh,
    [theme.breakpoints.down('xs')]: {
      padding: 0,
      margin: 0,
    },
  },
  content: {
    transition: 'margin .5s',
  },
});

const StickyCard = ({ classes, innerZ, cardProps, render, ...props }) => (
  <Sticky innerZ={innerZ} {...props}>
    {status => (
      <CallToActionArea
        {...cardProps}
        className={classNames(
          classes.animContainer,
          status.status === Sticky.STATUS_FIXED ? classes.animContainerSticky : classes.animContainerNormal
        )}
      >
        {render(status.status === Sticky.STATUS_FIXED)}
      </CallToActionArea>
    )}
  </Sticky>
);

StickyCard.propTypes = {
  ...Sticky.propTypes,
  children: PropTypes.node,
  classes: PropTypes.object,
  innerZ: PropTypes.number,
  cardProps: PropTypes.object,
  render: PropTypes.func,
};

StickyCard.defaultProps = {
  innerZ: 2000,
};
export default withStyles(styles)(StickyCard);
