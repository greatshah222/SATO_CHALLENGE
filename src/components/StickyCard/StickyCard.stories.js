import React from 'react';
import { storiesOf } from '@storybook/react';
import { withBackgrounds } from '@storybook/addon-backgrounds';
import { muiTheme } from 'storybook-addon-material-ui';
import theme from '../../theme';

import StickyCard from './index';
import Box from '../Box';
import Text from '../Text';
import PageContent from '../PageContent';
export const bg = withBackgrounds([{ name: 'light grey', value: '#f2f2f2', default: true }]);

storiesOf('StickyCard', module)
  .addDecorator(bg)
  .addDecorator(muiTheme(Object.values(theme)))
  .addWithJSX('default', () => (
    <div style={{ display: 'flex', flexDirection: 'column', flex: '1 0 auto' }}>
      <div style={{ width: '100%', height: '200px', background: 'orange' }} />
      <PageContent contentFullWidth>
        <Box variant={'pageSection'} negativeTopMargin>
          <StickyCard
            render={isFixed => (
              <React.Fragment>
                {!isFixed && (
                  <Text variant={'h3'} align={'center'}>
                    Show only if not fixed
                  </Text>
                )}
                <Text variant={'h1'} align={'center'}>
                  Sticky content
                </Text>
              </React.Fragment>
            )}
          />
        </Box>
      </PageContent>
      <PageContent>
        <div style={{ width: '100%', height: '1200px', background: 'inherit' }} />
      </PageContent>
    </div>
  ));
