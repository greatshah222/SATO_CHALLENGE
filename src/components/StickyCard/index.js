import StickyCard from './StickyCard';
export default StickyCard;
