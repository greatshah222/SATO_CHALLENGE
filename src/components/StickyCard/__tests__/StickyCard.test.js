import React from 'react';
import { shallow } from 'enzyme';
import StickyCard from '../StickyCard';

describe('StickyCard', () => {
  it('renders correctly', () => {
    const tree = shallow(<StickyCard />);
    expect(tree).toMatchSnapshot();
  });
});
