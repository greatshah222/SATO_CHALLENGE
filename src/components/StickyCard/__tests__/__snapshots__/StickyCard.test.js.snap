// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StickyCard renders correctly 1`] = `
<StickyCard
  classes={
    Object {
      "animContainer": "StickyCard-animContainer-1",
      "animContainerNormal": "StickyCard-animContainerNormal-2",
      "animContainerSticky": "StickyCard-animContainerSticky-3",
      "content": "StickyCard-content-4",
    }
  }
  innerZ={2000}
/>
`;
